drud/ddev
# feat: add DDEV Fig spec generation

## The Problem/Issue/Bug:

This FR automatically generates a [Fig spec](https://fig.io/docs/getting-started) for [Fig](https://fig.io/), as part of the `ddev_gen_autocompletes` command, so it does so as part of the automated build process.

This brings full terminal autocomplete for all Fig commands, with complete inline documentation. The resulting file does nothing on its own, but then you can either [push the spec to the autocomplete repo](https://fig.io/docs/guides/integrating/push-to-autocomplete-action) or [push it to Fig's cloud](https://fig.io/docs/guides/autocomplete-for-internal-tools#4-publish-your-spec).

Then it will be available to everyone who uses Fig.

## How this PR Solves The Problem:

This PR allows us to easily submit a Fig autocomplete for DDEV, so that everyone who uses Fig will benefit.

## Manual Testing Instructions:

Run:
```
make completions
cat .gotmp/bin/completions/ddev_fig_spec.ts
```
To see the resulting generated Fig Spec Typescript file.

## Automated Testing Overview:

No tests are introduced, as I saw no testing being done on the bash/zsh/etc autocomplete files, and the resulting file is just a generated file that does nothing until submitted as mentioned above.

## Related Issue Link(s):

It uses an [officially provided module](https://fig.io/docs/guides/integrating/integrations/cobra) for Cobra to do the work.

## Release/Deployment notes:

Nothing needs to be changed for release/deployment.

<a href="https://gitpod.io/#https://github.com/drud/ddev/pull/4216"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
